File: codewars/cpp/mumbling.cs
// merey's solution: 06.05.2026, 16:23 KBTU 3 corpus 702.
using System;
using System.Collections.Generic;
using System.Linq;

public class Accumul 
{
	public static string Accum(string s) 
  {
     char[] chars = s.ToCharArray();
     string b = "";
     int count = 0;
     for (int i = 1; i < s.Length + 1; i++)
     {
       for (int j = 1; j < i; j++)
       {
         b += char.ToLower(chars[count]);
       }
       
       b += "-";
       count++;
     }
    
    b = b[0..^1];
    b = char.ToUpper(b[0]) + b.Substring(1);
      
    for (int c = 0; c < b.Length; c++)
    {
      if (b[c] == '-') b = b.Substring(0, c + 1) + char.ToUpper(b[c + 1]) + b.Substring(c + 1);;
    }
    
    b = char.ToUpper(chars[0]) + b;
    
    return b;
  }
}

File: codewars/cpp/mumbling.cpp
#include <iostream>
#include <string>
using namespace std;
class Accumul
{
public:
    static char to_upper(char c){
      if (c >= int('a') && c <= int('z')){
        char result = char(int(c) - int('a') + int('A'));
          return result;
      }
      return c;
    }
    static char to_lower(char c){
        if (c >= int('A') && c <= int('Z')){
          char result = char(int(c) - int('A') + int('a'));
            return result;
        }
        return c;
      }
    static string accum(const string &s){
      string result = "";
      for (int i = 0; i < s.size(); ++i){
        char current_char = s[i];
        for (int j = 0; j <= i; ++j){
            if (j == 0) result = result + to_upper(current_char);
            else result = result + to_lower(current_char);
          }
        if (i != s.size() - 1) result = result + '-';
        }
      return result;
      }
  };